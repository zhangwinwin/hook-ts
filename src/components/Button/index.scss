@import '@resource/css/variable.scss';

.button {
  height: 30px;
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  background-color: $main-color;
  transition: background-color .4s;
  color: #fff;
  padding: 0 10px;

  &:hover {
    background-color: $main-color-sub;
  }

  &--isGray {
    background-color: $sub-color;

    &:hover {
      background-color: $sub-color-sub;
    }
  }
}