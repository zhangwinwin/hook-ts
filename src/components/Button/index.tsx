import React from 'react';
import cx from 'classnames';

import { Props } from 'src/types';
import './index.scss';

interface BtnProps extends Props {
  gray?: boolean;
  onClick(): void;
}

const Button = (props: BtnProps) => (
  <button
    onClick={props.onClick || null}
    className={cx('button', props.className, {
      'button--isGray': props.gray
    })}
  >
    {props.children}
  </button>
);

export default Button;
