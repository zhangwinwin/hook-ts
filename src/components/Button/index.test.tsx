/*= ==================
  Button.unit.test.tsx
 =================== */

import React from 'react';
import { shallow } from 'enzyme';
import Button from './index';

const handleClickBtn = () => {
  alert('Click button.');
};

describe('Button', () => {
  it('Renders button', () => {
    const button = shallow(<Button onClick={handleClickBtn}>Button</Button>);
    expect(button).toMatchSnapshot();
  });
});
