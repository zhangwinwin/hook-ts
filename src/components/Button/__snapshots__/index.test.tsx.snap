// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders button 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  Button
</button>
`;
